src/main/scala/firrtl/Visitor.scala

```diff
@@ -105,14 +105,29 @@ class Visitor(infoMode: InfoMode) extends FIRRTLBaseVisitor[FirrtlNode] {
         if (ctx.moduleBlock() != null)
           visitBlock(ctx.moduleBlock())
         else EmptyStmt)
-      case "extmodule" => ExtModule(info, ctx.id.getText, ctx.port.map(visitPort))
+      case "extmodule" =>
+        val defname = if (ctx.defname != null) ctx.defname.id.getText else ctx.id.getText
+        val ports = ctx.port map visitPort
+        val params = ctx.parameter map visitParameter
+        ExtModule(info, ctx.id.getText, ports, defname, params)
     }
   }
 
   private def visitPort[FirrtlNode](ctx: FIRRTLParser.PortContext): Port = {
     Port(visitInfo(Option(ctx.info), ctx), ctx.id.getText, visitDir(ctx.dir), visitType(ctx.`type`))
   }
 
+  private def visitParameter[FirrtlNode](ctx: FIRRTLParser.ParameterContext): Param = {
```

Comment:
That's more or less correct. FIRRTL.g4 is used by ANTLR to generate a Java lexer and parser. The parser returns a concrete syntax tree which we then "visit" to turn into an abstract syntax tree (as described in IR.scala). I could box everything here with Option and then match on (Some(blah), None, ...) but given that we are interacting with Java here I decided to just go ahead and deal with the nulls.